clients/csrf/csrf_token_manager_impl.go

```diff
@@ -0,0 +1,77 @@
+package csrf
+
+import (
+	"net/http"
+)
+
+const XCsrfHeader = "X-Csrf-Header"
+const XCsrfToken = "X-Csrf-Token"
+const CsrfTokenHeaderRequiredValue = "Required"
+
+type CsrfTokenManagerImpl struct {
+	request          *http.Request
+	transport        *Transport
+	csrfTokenFetcher CsrfTokenFetcher
+}
+
+func NewCsrfTokenManagerImpl(transport *Transport, request *http.Request, csrfTokenFetcher CsrfTokenFetcher) *CsrfTokenManagerImpl {
+	return &CsrfTokenManagerImpl{request: request, transport: transport, csrfTokenFetcher: csrfTokenFetcher}
+}
+
+func (c *CsrfTokenManagerImpl) setCsrfToken() error {
+	if c.request == nil || !c.isProtectionRequired(c.request, c.transport) {
+		return nil
+	}
+	err := c.initializeToken(false, getFetchNewTokenUrl(c.request))
+	if err != nil {
+		return err
+	}
+
+	c.updateCurrentCsrfTokens(c.request, c.transport)
+
+	return nil
+}
+
+func (c *CsrfTokenManagerImpl) initializeToken(force bool, url string) error {
```

Comment:
Why is the force needed?

Reply:
It's needed when we've initialized a csrf token and a new one is needed.